bridge/src/main/java/io/stargate/bridge/service/BridgeService.java

```diff
@@ -88,7 +94,27 @@ public void executeBatch(Batch batch, StreamObserver<Response> responseObserver)
   public void describeKeyspace(
       Schema.DescribeKeyspaceQuery request,
       StreamObserver<Schema.CqlKeyspaceDescribe> responseObserver) {
-    SchemaHandler.describeKeyspace(request, persistence, responseObserver);
+
+    String decoratedName =
+        persistence.decorateKeyspaceName(request.getKeyspaceName(), GrpcService.HEADERS_KEY.get());
+
+    Keyspace keyspace = persistence.schema().keyspace(decoratedName);
+    if (keyspace == null) {
+      responseObserver.onError(
+          Status.NOT_FOUND.withDescription("Keyspace not found").asException());
+    } else if (request.hasHash() && request.getHash().getValue() == keyspace.hashCode()) {
+      // Client already has the latest version, don't resend
+      responseObserver.onNext(EMPTY_KEYSPACE_DESCRIPTION);
+      responseObserver.onCompleted();
+    } else {
+      try {
+        Schema.CqlKeyspaceDescribe description = SchemaHandler.buildKeyspaceDescription(keyspace);
```

Comment:
We could cache this to avoid recomputing it for every client. It's a bit more complicated because we'll need to invalidate the cache when the hash changes, so overall I'm not sure if it's worth it.

Reply:
I think it's definitely worth it.. Here are some reasons why:

* first there will be multiple instances of same REST services, meaning that getting schema only once per version is not actually true, if the same tenant hits another instance, it has to go and get it
* second, with the optimization on the bridge you actually allow the REST side service to have no caching at all.. This can give a significant relaxation wrt to memory.. I know this is not the case for the GraphQL because caching allows you to build the GrapQL schema, but maybe for other services it is.. Maybe I am wrong here..